android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java

```diff
@@ -519,15 +519,25 @@ public boolean onEditorAction(TextView v, int actionId, KeyEvent keyEvent)
 			return true;
 		}
 
-		//This is to prevent 'return' event from being fired twice when return key is hit. In other words, when return key is clicked,
-		//this callback is triggered twice (except for keys that are mapped to EditorInfo.IME_ACTION_NEXT or EditorInfo.IME_ACTION_DONE). The first check is to deal with those keys - filter out
-		//one of the two callbacks, and the next checks deal with 'Next' and 'Done' callbacks, respectively.
-		//Refer to TiUIText.handleReturnKeyType(int) for a list of return keys that are mapped to EditorInfo.IME_ACTION_NEXT and EditorInfo.IME_ACTION_DONE.
-		if (actionId == EditorInfo.IME_ACTION_NEXT || actionId == EditorInfo.IME_ACTION_DONE) {
+		if (tv.getMaxLines() == 1) {
```

Comment:
This should use the `this.field` property like in the other PR.
The rest looks fine.